src/snowflake/snowpark/_internal/compiler/query_generator.py

```diff
@@ -89,8 +89,12 @@ def generate_queries(
             plan_queries = get_snowflake_plan_queries(
                 snowflake_plan, self.resolved_with_query_block
             )
-            queries.extend(plan_queries[PlanQueryType.QUERIES])
-            post_actions.extend(plan_queries[PlanQueryType.POST_ACTIONS])
+            for query in plan_queries[PlanQueryType.QUERIES]:
```

Comment:
I don't think we will need this, the queries and post actions for each plan should be disjoint

Reply:
without this change, when `_query_compilation_stage_enabled=True` and `cte_optimization_enabled=True`, duplicate queries re-arise.

Reply:
I see your point. You are right, we shouldn't de-duplicate at this point. I think there is another bug which is adding duplicate queries with new cte compilation stage. I'll try to find it.